packages/reports/src/reportBuilder/fetch/functions/dataGroups.ts

```diff
@@ -0,0 +1,57 @@
+import { Aggregator } from '../../../aggregator';
+import { FetchReportQuery } from '../../../types';
+import { FetchResponse } from '../types';
+import { Row } from '../../types';
+
+type DataGroupsFetchParams = {
+  programCode: string;
+};
+
+const fetchEvents = async (
+  aggregator: Aggregator,
+  query: FetchReportQuery,
+  params: DataGroupsFetchParams,
+): Promise<FetchResponse> => {
+  const { programCode } = params;
+  const { organisationUnitCode, startDate, endDate } = query;
+  const response = (await aggregator.fetchEvents(
+    programCode,
+    {
+      useDeprecatedApi: false,
+      dataServices: { isDataRegional: true },
+      organisationUnitCodes: [organisationUnitCode],
+      startDate,
+      endDate,
+    },
+    {},
+  )) as Row[];
+  response.forEach(row => {
+    Object.entries(row.dataValues).forEach(([key, value]) => {
+      row[key] = value;
+    });
+    delete row.dataValues;
```

Comment:
We should somehow make sure that `key`s do not conflict with other event metadata keys (`eventDate` etc) This will most probably not be a problem in practice, since we use different naming conventions for those (upper SNAKE_CASE for data element keys, camelCase for metadata)

Reply:
Oh good point... Tbh I'm not really happy with how I'm handling events here... Perhaps we just leave them as they come from the Aggregator, and then have a named transform format the data correctly, eg.
```
{
  "fetch": { "dataGroups": ["HP02"] },
  "transform": [
    "expandDataGroups",
    ...
  ]
}
```

Since we don't need to be using events in the short term, I'm happy to make that decision at a later date. But yeah for the record I still don't know the best way to handle events...